pkg/gadgets/top/ebpf/tracer/tracer.go

```diff
@@ -325,7 +322,15 @@ func (t *Tracer) nextStats() ([]*types.Stats, error) {
 			CumulativeRunCount: cumulativeRunCount,
 			MapMemory:          totalMapMemory,
 			MapCount:           uint32(len(mapIDs)),
-		})
+		}
+
+		if t.enricher != nil {
+			// We are only interested in node here, so giving 0 as MountNsID should be
+			// fine.
```

Comment:
This seems a bit hacky to me. What about defining a new `EnrichNode` enricher that doesn't receive any argument?

Reply:
What about something like this https://github.com/inspektor-gadget/inspektor-gadget/commit/69d38c65d4c21a6dc49c7900f92658b9c2329a45?